src/pt/hentaiteca/src/eu/kanade/tachiyomi/extension/pt/hentaiteca/HentaiTeca.kt

```diff
@@ -1,20 +1,47 @@
 package eu.kanade.tachiyomi.extension.pt.hentaiteca
 
+import android.app.Application
+import androidx.preference.PreferenceScreen
+import eu.kanade.tachiyomi.lib.randomua.addRandomUAPreferenceToScreen
+import eu.kanade.tachiyomi.lib.randomua.getPrefCustomUA
+import eu.kanade.tachiyomi.lib.randomua.getPrefUAType
+import eu.kanade.tachiyomi.lib.randomua.setRandomUserAgent
 import eu.kanade.tachiyomi.multisrc.madara.Madara
 import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.ConfigurableSource
+import okhttp3.Headers
 import okhttp3.OkHttpClient
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
 import java.text.SimpleDateFormat
 import java.util.Locale
 import java.util.concurrent.TimeUnit
 
-class HentaiTeca : Madara(
-    "Hentai Teca",
-    "https://hentaiteca.net",
-    "pt-BR",
-    SimpleDateFormat("MMMMM dd, yyyy", Locale("pt", "BR")),
-) {
+class HentaiTeca :
+    Madara(
+        "Hentai Teca",
+        "https://hentaiteca.net",
+        "pt-BR",
+        SimpleDateFormat("dd/MM/yyyy", Locale("pt", "BR")),
+    ),
+    ConfigurableSource {
+
+    private val preferences = Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
 
     override val client: OkHttpClient = super.client.newBuilder()
+        .setRandomUserAgent(
+            preferences.getPrefUAType(),
+            preferences.getPrefCustomUA(),
+        )
         .rateLimit(1, 2, TimeUnit.SECONDS)
         .build()
+
+    override fun headersBuilder(): Headers.Builder = super.headersBuilder()
+        .set("User-Agent", "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:123.0) Gecko/20100101 Firefox/123.0")
```

Comment:
why hardcode the useragent?

Reply:
> why hardcode the useragent?

Maybe can be removed, but users will have to change the global UA in the app, as the J2K UA is old and cloudflare blocks it

Reply:
> > why hardcode the useragent?
> 
> Maybe can be removed, but users will have to change the global UA in the app, as the J2K UA is old and cloudflare blocks it

Removed